### PR TITLE
Set the session handling tests to run in parallel

### PR DESCRIPTION
**Pull request by viorelaioia-zz**

### CODE REVIEW

**Comment by johngian** [comment]
Architecture-wise it is not a good idea to have a shared state between tests because this has as a result them not to be atomic. That said i don't have a solid answer on how to synchronise the tests when running them in parallel.

One more scrapy idea could be to:

* Use an OTP code
* Wait for the result
* If login works we are happy \o/
* If not we check the result message
* If it is related to an OTP duplicate we retry

File: pages/two_factor_authentication.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        self.selenium.switch_to_default_content()
+
+    @property
+    def is_error_message_displayed(self):
+        if self.is_element_present(*self._duo_iframe_locator):
+            self.selenium.switch_to_frame('duo_iframe')
+            is_message_shown = self.selenium.find_element(*self._error_message_locator).is_displayed()
+            self.selenium.switch_to_default_content()
+            return is_message_shown
+        else:
```
Comment (johngian) [inline pages/two_factor_authentication.py]: No need for else. A more pythonic way to write this is:

```
if foo:
    return "bar"
return "foobar"
```

File: tests/conftest.py [hunk 2]
```diff
@@ -43,3 +43,15 @@ def access_token(token):
...
     return json.loads(response.content)['access_token']
+
+
+@pytest.fixture
+def file_content(filename):
+    return open(filename, 'r').read()
+
+
+@pytest.fixture
+def write_in_file(filename, content):
```
Comment (johngian) [inline tests/conftest.py]: I don't see any reference to this. Do you still need this?

File: tests/conftest.py [hunk 3]
```diff
@@ -43,3 +43,15 @@ def access_token(token):
     }
     response = requests.post(token['url'], data=json.dumps(data), headers=headers)
     return json.loads(response.content)['access_token']
+
+
+@pytest.fixture
+def file_content(filename):
```
Comment (johngian) [inline tests/conftest.py]: Same here

**Review by johngian — APPROVED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (viorelaioia-zz) [reply tests/conftest.py]: I forgot to delete this. Thanks!

**Pull request merged**